### ISSUE TITLE
`useQuery` with `enabled` prop causing infinite loop with local Suspense/ErrorBoundary components

### ISSUE THREAD

**Issue by erikshestopal** [blitz-js/blitz#2538]
### What is the problem?

When the using the `useQuery` hook with the `enabled` property that is controlled through a prop passed into the component, an infinite loop is caused when the the query resolver throws an error. 

I have created a [Codesandbox](https://codesandbox.io/s/heuristic-dust-ws2ro?file=/app/pages/index.tsx) with a reproduction.

### Paste all your error logs here:

```
N/A
```

### Paste all relevant code snippets here:

```tsx
import { Suspense, useState } from "react";
import { BlitzPage, useQuery, ErrorBoundary } from "blitz";
import Layout from "app/core/layouts/Layout";
import getCurrentUser from "app/users/queries/getCurrentUser";

// Query resolver
import { Ctx } from "blitz";

export default async function getCurrentUser(_ = null, { session }: Ctx) {
  throw new Error("I will cause an infinite loop");
}


const DataFetch = ({ count }) => {
  useQuery(getCurrentUser, null, {
    // This enabled prop is causing the problem
    enabled: count > 1
  });

  return (
    <div>
      Click the button above twice to cause infinite loop in{" "}
      <code>useQuery</code>
    </div>
  );
};

const Home: BlitzPage = () => {
  const [count, setCount] = useState(0);

  return (
    <div className="container">
      <main>
        <h1>{count}</h1>
        <button onClick={() => setCount((count) => count + 1)}>
          Cause infinite loop
        </button>
        <ErrorBoundary fallback={<div>errored</div>}>
          <Suspense fallback={<div>loading...</div>}>
            <div style={{ marginTop: "1rem" }}>
			  { /* count is used to control when the query hook runs */}	
              <DataFetch count={count} />
            </div>
          </Suspense>
        </ErrorBoundary>
      </main>
    </div>
  );
};

Home.suppressFirstRenderFlicker = true;
Home.getLayout = (page) => <Layout title="Home">{page}</Layout>;

export default Home;

```

### What are detailed steps to reproduce this?

Please see Codesandbox. 

### Run `blitz -v` and paste the output here:

```
macOS Big Sur | darwin-x64 | Node: v15.12.0

blitz: 0.34.3 (global)
blitz: 0.38.2 (local)

  Package manager: yarn
  System:
    OS: macOS 11.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
    Memory: 1.16 GB / 32.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 15.12.0 - ~/.nvm/versions/node/v15.12.0/bin/node
    Yarn: 1.22.10 - ~/.nvm/versions/node/v15.12.0/bin/yarn
    npm: 7.6.3 - ~/.nvm/versions/node/v15.12.0/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  npmPackages:
    @prisma/client: 2.25.0 => 2.25.0
    blitz: 0.38.2 => 0.38.2
    prisma: 2.25.0 => 2.25.0
    react: alpha => 18.0.0-alpha-d7dce572c
    react-dom: alpha => 18.0.0-alpha-d7dce572c
    typescript: ~4.2 => 4.2.4
```

**Comment by erikshestopal** [comment]
Probably a duplicate of https://github.com/blitz-js/blitz/issues/2368 now that I look through existing issues 🤔

**Comment by flybayer** [comment]
Thank you! I'm able to reproduce and am working on fixing it.

**Comment by flybayer** [comment]
Well, this is a bug in react-query itself: https://github.com/tannerlinsley/react-query/issues/2434

So we'll have to wait on a fix there. Anyone's welcome to contribute a fix there.

**Comment by erikshestopal** [comment]
Thanks @flybayer!
